File: boj/1212.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        int len = s.length();
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < len; ++i) {
            int num = s.charAt(i) - 48;
            StringBuilder temp = new StringBuilder();
            for (int j = 0; j < 3; ++j) {
                temp.append(num % 2);
                num /= 2;
                if (i == 0 && num == 0) break;
            }
            temp.reverse();
            ans.append(temp);
        }
        System.out.println(ans);
    }
}

File: boj/1212.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main(void) {
    ios_base::sync_with_stdio(false);
    string s; cin >> s;
    int len = s.length();
    string ans;
    for (int i = 0; i < len; ++i) {
        int num = s[i] - 48;
        string temp;
        for (int j = 0; j < 3; ++j) {
            temp += (char)(num % 2) + 48;
            num /= 2;
            if (i == 0 && num == 0) break;
        }
        reverse(temp.begin(), temp.end());
        ans += temp;
    }
    cout << ans << "\n";
    return 0;
}
